exthttp/exthttp.go

```diff
@@ -59,9 +61,27 @@ func (w *LoggingHttpResponseWriter) WriteHeader(statusCode int) {
 	w.delegate.WriteHeader(statusCode)
 }
 
+func RequestTimeoutHeaderAware(next func(w http.ResponseWriter, r *http.Request)) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		timeout := r.Header.Get("Request-Timeout")
+		if timeout == "" {
+			r.Header.Get("X-Request-Timeout")
+		}
+		if timeout != "" {
+			timeoutValue, err := strconv.ParseFloat(timeout, 32)
+			if err == nil {
+				log.Trace().Msgf("Using handler timeout %.1fs", timeoutValue)
+				http.TimeoutHandler(http.HandlerFunc(next), time.Duration(timeoutValue*1000)*time.Millisecond, "Request timed out.").ServeHTTP(w, r)
```

Comment:
**Question:** Does this also set a deadline on the `context.Context` of the request?